File: solutions/2022/main.cpp
/* ************************************************************************
> File Name:     main.cpp
> Author:        Sean
> Project:    LeetCode
> Created Time:  Sat Jan  1 11:16:18 2022
> Description:   
 ************************************************************************/
#include <vector> 
#include <iostream> 
#include <algorithm> 
#include <gtest/gtest.h> 

using namespace std; 

class Solution {
public:
    vector<vector<int>> construct2DArray(vector<int>& original, int m, int n) {
        if (original.size() != m*n) {
            vector<vector<int>> empty;
            return empty;
        };
        vector<vector<int>> result(m, vector<int>(n, 0));
        for (int i = 0; i < m; ++i) {
            auto start = original.begin()+i*n;
            auto end = original.begin()+(i+1)*n;
            copy(start, end, result[i].begin());
        }
        return result;
    }
};

TEST(Test2022, SimpleTest) {
    vector<int> original{
        1,2,3,4
    };
    const int m = 2, n = 2; 
    Solution s; 
    vector<vector<int>> result = s.construct2DArray(original, m, n); 
    for (auto& r : result) {
        for (auto& c : r)
            cout << c << " ";
        cout << endl; 
    }
}

int main() {
    testing::InitGoogleTest();
    return RUN_ALL_TESTS();
}

File: solutions/2022/main.py
from typing import List 

class Solution:
    def construct2DArray(self, original: List[int], m: int, n: int) -> List[List[int]]:
        if len(original) != m*n:
            return []
        result = [[0]*n for _ in range(m)]
        for i in range(m):
            result[i] = original[i*n:(i+1)*n]
        return result

if __name__ == "__main__": 
    original = [1,2,3,4] 
    m = 2 
    n = 2
    s = Solution()
    result = [[1,2],[3,4]]
    assert s.construct2DArray(original, m, n) == result
